docs/prow/release-checklist.md

```diff
@@ -0,0 +1,89 @@
+# Release checklist
+
+This document describes how to execute release of Kyma using prow.
+
+## Release preparation
+
+This part can be done at any point of time. 
+This part applies only for new major and minor versions.
+
+1. Define release jobs in test-infra repository
+ - for every component
+ - for every tool
+ - kyma-integration
+ - kyma-gke-integration
+ - kyma-artifacts
+ - kyma-installer
+
+To see all release jobs for release 0.6, look for job names with prefix: `pre-rel06`.
+Every job name has to have unique name, so remember about prefixing it with `pre-rel<XY>`
+
+2. Ensure that tests for jobs exist
+3. Define branch protection rules for the release branch.
+For release-0.6 definition was following:
+```
+release-0.6:
+  protect: true
+  required_status_checks:
+    contexts:
+      - pre-rel06-kyma-integration
+      - pre-rel06-kyma-gke-integration
+      - pre-rel06-kyma-artifacts
+      - pre-rel06-kyma-installer
+```
+
+
+## Release
+
+1. Create release branch in test-infra repository.
+This part applies only for new major and minor versions.
+The name of this branch should follow the `release-x.y` pattern, such as `release-0.6`.
+
+2. Ensure that `RELEASE_VERSION` file contains version to be created.
```

Comment:
Ensure that `prow/RELEASE_VERSION`...
and maybe add some examples like: `0.6.0-rc1`, `0.6.0` etc - it suggests how content file should looks like and also there should be every version no only major and minor